database/sql/V012__extend_orders_for_contract_upgrade.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TYPE FundLocation AS ENUM ('owner', 'vault_internal', 'vault_external');
+
+ALTER TABLE orders
+    ADD COLUMN version_id INT,
```

Comment:
Maybe you can use `NOT NULL WITH DEFAULT` here instead of doing 3 separate SQL statements.

Reply:
Did you read the description of this PR or [this comment](https://github.com/gnosis/gp-v2-services/issues/898#issuecomment-887699878) in the issue? 

I will highlight the reason I have chosen against this, but like the description says, the first commit here is an implementation with this suggestion if we would prefer to go this way.

> Option 1: Add non-null columns with default values. This would mean that we will forever have to ensure we don't accidentally insert without specifying these values explicitly (this would be especially important for the settlement version, since the default would be the old contract version. I am not a fan of the default value for the contract version being set to the old contracts

Later on we discuss Option 3 which I quite like (as a mix of the two)

> Do Option 2 for settlement_version and Option 1 for the other two columns (essentially making owner) the default value for balance to and from fields.

Reply:
I read it but maybe misunderstood what you meant by it. I missed that with my suggestion all future rows inserted would also contain the default value - agree that we don't want that!

Makes perfect sense then (maybe we can add a small comment in the code)?

Reply:
After googling, I found an alternative with 2 queries would be to drop the default later on:

> ALTER TABLE mytable ALTER COLUMN mycolumn DROP DEFAULT;

But I'm also fine with the existing solution.

Reply:
Will add a comment. I think the drop default solution is cool, but we would have to remember to do that at a later point.

Reply:
Couldn't it simply happen inside this migration script instead of statement 2 & 3?

Reply:
Drop default solution has an advantage that DB won't have to check that all values are not NULL.

Reply:
Only thing I would like to check is that the existing records are actually updated with the default values.

Reply:
Works like a charm!